app/service/collection.py

```diff
@@ -166,15 +172,25 @@ def create(
     :raises ValidationError: raised should the import_id be invalid.
     :return str: The new import_id for the collection.
     """
-    if context is not None:
-        context.error = "Error when creating collection"
-
-    return collection_repo.create(db, collection, user.org_id)
 
+    if db is None:
+        db = db_session.get_db()
 
-@db_session.with_transaction(__name__)
+    try:
+        import_id = collection_repo.create(db, collection, user.org_id)
+        if len(import_id) == 0:
+            db.rollback()
+        return import_id
+    except Exception as e:
+        # db.rollback()
```

Comment:
Commented out code?

Reply:
hmmm -- thought that was restored - let me fix